runtime/util/vmargs.c

```diff
@@ -76,8 +76,46 @@
 
 #define LARGE_STRING_BUF_SIZE 256
 
+struct JavaLauncherOption {
+	char *optionName;
+	int optionNameLength;
+};
+
+/* options in this table will be not treated as a user Argument 
+Note, these options need to be in lexical order (ASCII values) for the binary search function to properly work */
+static const struct JavaLauncherOption JavaLauncherOptionTable[] = {
+	{"-Djava.class.path=", strlen("-Djava.class.path=")},
+	{"-Dsun.java.command=", strlen("-Dsun.java.command=")},
+	{"-Dsun.java.launcher.pid=", strlen("-Dsun.java.launcher.pid=")},
+	{"-Dsun.java.launcher=", strlen("-Dsun.java.launcher=")},/* Note, 'launcher=' is placed after .launcher.pid` since '=' has a higher ASCII value than '.' */
+	{"-Xjcl:", strlen("-Xjcl:")},
+};
+
+#define ignoredOptionTableEnd (sizeof(JavaLauncherOptionTable) / sizeof(struct JavaLauncherOption) - 1)
+
 static char * getStartOfOptionValue(J9VMInitArgs* j9vm_args, IDATA element, const char *optionName);
 
+UDATA
+isGeneratedArgument(const char *optionName) {
+	IDATA start = 0;
+	IDATA end = ignoredOptionTableEnd;
+	IDATA mid = 0;
+	while (start <= end) {
+		mid = (start + end) / 2;
+		IDATA optionCompare = strncmp(optionName, JavaLauncherOptionTable[mid].optionName, JavaLauncherOptionTable[mid].optionNameLength);
+		if (optionCompare == 0) {
+			/* The search function found the given option name in the table. */
+			return TRUE;
```

Comment:
A single return point is preferred. Declare and return a status variable; initialize it to FALSE, if you find a match set it to TRUE and break the loop.

Reply:
I was thinking of simply breaking the loop `if (optionCompare == 0)` and return `(optionCompare == 0)? TRUE : FALSE;` 
This way, we don't need to introduce a new variable for the return status

Reply:
You could do that, provided you initialize the variable suitably. 
Style comments: 
- put the rvalue (e.g. a constant) on the left side of a `==` or `!=` for safety.
- `(optionCompare == 0)? TRUE : FALSE` is redundant, since `(optionCompare == 0)` evaluates to `TRUE` or `FALSE`.

Reply:
I've decided to switch to a linear search as requested by Keith for the following:
-It might be overkill to implement a binary search since were only reducing the number of loops we’re doing by 2 (MAX)
-With a linear search, the only variable we need to declare is a pointer, which is much cleaner than having a start/end/mid variable.
-The table will not need to be in Lexical Order (edited)

Reply:
Regarding your comment earlier about having a single return point, would you still like me to implement that? I find that it might not be as effective for the following reasons:

-If we decide to have a boolean variable storing the return value upon exiting the loop, then we introduce another variable in the function
-If we don't introduce a new variable, then the only other solution is to return another call of `strncmp()`...